src/utils/bridge.ts

```diff
@@ -178,14 +178,34 @@ export async function sendAcrossDeposit(
   }: AcrossDepositArgs
 ): Promise<ethers.providers.TransactionResponse> {
   const config = getConfig();
-  const spokePool = config.getSpokePool(fromChain);
   const provider = getProvider(fromChain);
-  const code = await provider.getCode(spokePool.address);
-  if (!code) {
+
+  // We send native tokens to the SpokePoolVerifier contract if enabled for the chain.
+  const shouldUseSpokePoolVerifier =
+    config.enabledSpokePoolVerifierChains.has(fromChain) && isNative;
+
+  const spokePool = config.getSpokePool(fromChain);
+  const spokePoolVerifier = shouldUseSpokePoolVerifier
+    ? config.getSpokePoolVerifier(fromChain)
+    : undefined;
+
+  const [spokePoolCode, spokePoolVerifierCode] = await Promise.all([
+    provider.getCode(spokePool.address),
+    spokePoolVerifier
+      ? await provider.getCode(spokePoolVerifier.address)
```

Comment:
Do we need this additional await? Should the promise be resolved in the `Promise.all( )`?
```suggestion
      ? provider.getCode(spokePoolVerifier.address)
```

Reply:
Changed https://github.com/across-protocol/frontend-v2/pull/861/commits/720e5fbdd7c67bc5bda8afaf6435040a16b2f898